File: 2024/Java/Day3Part1.java
import java.io.*;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Day3Part1 {
    public static void main(String[] args) throws IOException {
        try (BufferedReader reader = Files.newBufferedReader(Path.of("2024/Input/day3.txt"))) {
            String line;
            int sum = 0;
            while((line = reader.readLine()) != null) {
                Pattern pattern = Pattern.compile("mul\\((\\d+),(\\d+)\\)");
                Matcher match = pattern.matcher(line);
                while(match.find()) {
                    int firstNumber = Integer.parseInt(match.group(1));
                    int secondNumber = Integer.parseInt(match.group(2));
                    sum += firstNumber * secondNumber;
                }
            }
            System.out.println(sum);
        }
    }
}

File: 2024/Python/day3part1.py
import re

sum = 0
pattern = r'mul\(\d{1,3},\d{1,3}\)'
file = open("2024/Input/day3.txt")
for line in file:
    line = re.findall(pattern, line)
    line = "".join(line)
    line = re.findall(r'\(\d{1,3},\d{1,3}\)', line)
    for numbers in line:
        numbers = numbers[1:len(numbers) - 1]
        firstNumber = numbers.split(',')[0]
        secondNumber = numbers.split(',')[1]
        sum = sum + (int(firstNumber) * int(secondNumber))
print(sum)